Test/runtime/file-io/AbstractFileIO.dfy

```diff
@@ -0,0 +1,27 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+include "../../libraries/src/Wrappers.dfy"
+
+abstract module AbstractFileIO {
+  import opened Wrappers
+
+  export provides ReadBytesFromFile, WriteBytesToFile, Wrappers
+
+  method ReadBytesFromFile(path: string) returns (res: Result<seq<bv8>, string>) {
+    var isError, bytesRead, errorMsg := INTERNAL_ReadBytesFromFile(path);
+    return if isError then Failure(errorMsg) else Success(bytesRead);
+  }
+
+  method WriteBytesToFile(path: string, bytes: seq<bv8>) returns (res: Result<(), string>)
+  {
+    var isError, errorMsg := INTERNAL_WriteBytesToFile(path, bytes);
+    return if isError then Failure(errorMsg) else Success(());
+  }
+
+  method INTERNAL_ReadBytesFromFile(path: string)
+    returns (isError: bool, bytesRead: seq<bv8>, errorMsg: string)
```

Comment:
I concur with wondering about the correct sign of this output.

Reply:
I don't have a strong opinion either way, but since this an internal API (users of the file I/O API should never see/use this) I'm not too concerned either way.